lib/ansible/template/vars.py

```diff
@@ -105,6 +105,8 @@ def __getitem__(self, varname):
             value = None
             try:
                 value = self._templar.template(variable)
+            except AnsibleUndefinedVariable as e:
+                raise AnsibleUndefinedVariable(to_native(variable) + ': ' + e.message)
```

Comment:
Instead of doing this, can we simply re-raise the existing exception?  That way the traceback information stays intact:

``` python
except AnsibleUndefinedVariable as e:
    raise
except Exception as e:
    [....]
```

Reply:
Done